flytekit/common/tasks/sdk_dynamic.py

```diff
@@ -150,10 +157,28 @@ def _produce_dynamic_job_spec(self, context, inputs):
         visited_nodes = set()
         generated_ids = {}
         effective_failure_ratio = self._allowed_failure_ratio or 0.0
+
+        # TODO: This function needs to be cleaned up.
+        # The reason we chain these two together is because we allow users to not have to explicitly "yield" the
+        # node. As long as the subtask/lp/subwf has an output that's referenced, it'll get picked up.
         for sub_task_node in _itertools.chain(yielded_sub_tasks, upstream_nodes):
             if sub_task_node in visited_nodes:
                 continue
             visited_nodes.add(sub_task_node)
+            executable = sub_task_node.executable_sdk_object
+
+            # If the executable object that we're dealing with is registerable (ie, either an SdkRunnableLaunchPlan or
+            # an SdkWorkflow), then it should have the ability to give itself a name.  After assigning itself the name,
+            # also make sure the id is properly set according to current config values.
+            if isinstance(executable, _registerable.RegisterableEntity):
```

Comment:
this should hit for task, launch plan, and workflow since they're all registerable

Reply:
right, updated the comment.